### PR TITLE
Improve configuration API and support for initializers

### PR DESCRIPTION
**Pull request by aflatter**
Initializers like the following can now be used to configure repositories:

```ruby
Rails.application.configure do |app|
  app.config.rom.repos[:yaml] = {uri: 'yaml:///config'}
end
```

### CODE REVIEW

**Comment by solnic** [comment]
Do you think we could yield rom's config instead of the whole app? Or maybe that wouldn't be flexible enough?

File: lib/rom/rails/configuration.rb [hunk 1]
```diff
@@ -1,33 +1,36 @@
 module ROM
   module Rails
     class Configuration
-      attr_reader :config, :setup, :env
+      attr_reader :repos
```
Comment (solnic) [inline lib/rom/rails/configuration.rb]: It'd be nice to use `repositories` here. I know it's longer but that's how it's called in ROM. I'm probably guilty of using the word "repos" when chatting about it, please excuse me :)

Comment (aflatter) [inline lib/rom/rails/configuration.rb]: Sounds good!

**Comment by aflatter** [comment]
What would that look like? `Rails.application.config.rom.configure { |config| ... }`?

**Comment by solnic** [comment]
I thought about something like:

``` ruby
ROM::Rails.configure do |config|
  config.repositories[:cosmos] = { uri: "nasa://outerspace/galaxy" }
end
```

**Comment by aflatter** [comment]
Close enough?

```ruby
ROM::Rails::Railtie.configure do |config|
  config.repositories[:yaml] = {uri: 'yaml:///config'}
end
```

**Comment by solnic** [comment]
Can you tell why we can't do `ROM::Rails.configure`? Or you just think it's better to put it there on Railtie class?

**Comment by aflatter** [comment]
The latter. `Railtie` is a weird kind of singleton already and I don't want it to creep into other parts. Maybe we should actually move things out of `Railtie` and make it really lightweight. This would lessen the confusion around `config` having two meanings (The global `Rails.application.config` and Rom's subset `Rails.application.config.rom`).

**Comment by solnic** [comment]
Yes this sounds very reasonable. I'm really unexperienced when it comes to building railties. I put things together to make them work somehow which was a success in itself for me. It definitely needs some reorganization and more love.

**Pull request merged**